tracpro/contacts/models.py

```diff
@@ -97,10 +100,12 @@ def kwargs_from_temba(cls, org, temba_contact):
 
         org_region_uuids = [r.uuid for r in Region.get_all(org)]
         region_uuids = intersection(org_region_uuids, temba_contact.groups)
-        region = Region.objects.get(org=org, uuid=region_uuids[0]) if region_uuids else None
-
-        if not region:  # pragma: no cover
-            raise ValueError("No region with UUID in %s" % ", ".join(temba_contact.groups))
+        if len(region_uuids) > 0:
+            region = Region.objects.filter(org=org, uuid__in=region_uuids).first() if region_uuids else None
+            if not region:  # pragma: no cover
```

Comment:
Yes, So what I did was to create another exception that would tell us which
contact has not been created because they don't belong to any region. From
what I know about tracpro, is that the exception should still come because
we don't want contacts with no regions in. Except this time we can tell
from the logs who the contact is and we can attempt to investigate further
from there. Are we starting to let contacts in who don't have any regions?

On Fri, Apr 10, 2015 at 1:44 PM, Rowan Seymour <notifications@github.com>
wrote:

> In tracpro/contacts/models.py
> <https://github.com/ewheeler/tracpro/pull/1#discussion_r28135823>:
>
> > @@ -97,10 +100,12 @@ def kwargs_from_temba(cls, org, temba_contact):
> >
> >          org_region_uuids = [r.uuid for r in Region.get_all(org)]
> >          region_uuids = intersection(org_region_uuids, temba_contact.groups)
> > -        region = Region.objects.get(org=org, uuid=region_uuids[0]) if region_uuids else None
> > -
> > -        if not region:  # pragma: no cover
> > -            raise ValueError("No region with UUID in %s" % ", ".join(temba_contact.groups))
> > +        if len(region_uuids) > 0:
> > +            region = Region.objects.filter(org=org, uuid__in=region_uuids).first() if region_uuids else None
> > +            if not region:  # pragma: no cover
>
> Am not sure I understand what we're trying to do here. I think the
> original exception was occurring because the contact was not in any of the
> configured region groups, and region is a required field for a contact.
> region_uuids is the intersection of the set of all region group UUIDs,
> and of the incoming contact's group UUIDs. If region_uuids is not empty,
> then region will never be falsey. I may be missing something but I think
> all this does is change the exception.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ewheeler/tracpro/pull/1/files#r28135823>.
>